### PR TITLE
Bug Fix

### PR DESCRIPTION
**Pull request by Neche-Stephen**
Fixed bugs on removing unnecessary libraries, removing mock tests, and optimizing process meta function.

### CODE REVIEW

File: src/chatgpt-log.ts [hunk 1]
```diff
@@ -56,22 +59,25 @@ export class ChatGptLog {
...
-      userTimeZone: metaDetails?.data?.data?.userTimeZone,
-      memory: metaDetails?.data?.data?.memory,
-      machineId: metaDetails?.data?.data?.machineId,
-      env: metaDetails?.data?.data?.env,
+      totalTokenCount:  tokenCount,
+      inputTokenCost1k: modelCost?.tokensInCost,
+      outputTokenCost1k: modelCost?.tokensOutCost,
+      triggerSource: "",
+      // outputMode: metaDetails?.data?.data?.outputMode,
+      userId: uuidv4(),
```
Comment (justiceo) [inline src/chatgpt-log.ts]: The caller would provide this param. Let's leave it empty for now and also remove the uuid package.

File: src/chatgpt-log.ts [hunk 2]
```diff
@@ -56,22 +59,25 @@ export class ChatGptLog {
...
+      country: "",
+      operatingSystem: `${os.type()} ${os.release()}`,
```
Comment (justiceo) [inline src/chatgpt-log.ts]: Update to use the format "os.platform()/os.release()", os.platform() is much more precise than type()

File: spec/chatgpt.spec.ts [hunk 3]
```diff
@@ -0,0 +1,110 @@
+import { start } from "repl";
```
Comment (justiceo) [inline spec/chatgpt.spec.ts]: I don't see where `start` is used in this file. Also the repl package referenced here is not a declared dependency in package.json.

Remove from here and other places.

File: src/chatgpt-log.ts [hunk 4]
```diff
@@ -1,7 +1,10 @@
 import { start } from "repl";
```
Comment (justiceo) [inline src/chatgpt-log.ts]: Update your imports, here and elsewhere to follow the order

 1. system packages e.g. "os", "path"
 2. 3rd party libraries e.g. "node-machine-id"
 3. local dependencies e.g. "./log-entry"

File: spec/chatgpt.spec.ts [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+    condition: "Sunny",
+  };
+  return weatherData;
+}
+
+export async function useChatGptApi(prompt: string) {
+  flow.logPrompt(prompt, "user-input");
+
+  const requestData: ChatCompletionCreateParamsNonStreaming = {
+    model: "gpt-4o",
```
Comment (justiceo) [inline spec/chatgpt.spec.ts]: Update to use gpt-4o-mini

**Review by justiceo — CHANGES_REQUESTED** [review]

File: src/chatgpt-log.ts [hunk 6]
```diff
@@ -71,7 +71,7 @@ export class ChatGptLog {
       // outputMode: metaDetails?.data?.data?.outputMode,
       userId: 0,
       country: "",
-      operatingSystem: `${os.type()} ${os.release()}`,
+      operatingSystem: `${os.platform()} ${os.release()}`,
```
Comment (justiceo) [inline src/chatgpt-log.ts]: "/" for consistency

File: spec/chatgpt.spec.ts [hunk 7]
```diff
@@ -1,9 +1,21 @@
-import { flow } from "../src/flow";
-import OpenAI from "openai";
+// System packages
+
+// Third-party libraries
+
+// Local dependencies
+
+
+// System packages
```
Comment (justiceo) [inline spec/chatgpt.spec.ts]: Keep it in mind, no need to put it in the code :)

**Review by justiceo — CHANGES_REQUESTED** [review]

File: spec/chatgpt.spec.ts (hunk 7 above)
Comment (Neche-Stephen) [reply spec/chatgpt.spec.ts]: Oh dear. I forgot to remove that. lol

File: spec/gemini.spec.ts [hunk 8]
```diff
@@ -44,7 +48,7 @@ export async function useGeminiApi(prompt: string) {
   flow.logPrompt(prompt, "user-input");
 
   const requestData = {
-    model: "gemini-1.5-flash",
+    model: "models/gemini-1.5-pro",
```
Comment (justiceo) [inline spec/gemini.spec.ts:51]: Use flash

File: data/2024-11-01.jsonl
Comment (justiceo) [inline data/2024-11-01.jsonl:1]: Both jsonl files look outdated based on current implementation.

**Review by justiceo — APPROVED** [review]
See comments

File: data/2024-11-01.jsonl
Comment (Neche-Stephen) [reply data/2024-11-01.jsonl:1]: Yea, the PR has been around for a long time. Subsequent PRS will have more recent entries

**Pull request merged**